File: entities/account.go
package entities

import "time"

type TransferResult struct {
	From Account `json:"from"`
	To   Account `json:"to"`
}

type Account struct {
	ID           uint      `json:"id" gorm:"primaryKey"`
	Name         string    `json:"name" gorm:"size:255;not null"`
	Surname      string    `json:"surname" gorm:"size:255;not null"`
	Email        string    `json:"email" gorm:"size:255;uniqueIndex;not null"`
	Phone        *string   `json:"phone" gorm:"size:50"`
	AddressLine1 string    `json:"addressLine1" gorm:"column:address_line1;size:255;not null"`
	AddressLine2 *string   `json:"addressLine2" gorm:"column:address_line2;size:255"`
	City         string    `json:"city" gorm:"size:100;not null"`
	Postcode     string    `json:"postcode" gorm:"size:20;not null"`
	Country      string    `json:"country" gorm:"size:100;not null"`
	Balance      int64     `json:"balance" gorm:"default:0;not null"`
	Currency     string    `json:"currency" gorm:"size:3;default:'EUR';not null"`
	CreatedAt    time.Time `json:"createdAt" gorm:"column:created_at"`
	UpdatedAt    time.Time `json:"updatedAt" gorm:"column:updated_at"`
}

File: ts_solution/src/entities/Account.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
} from "typeorm";

@Entity("accounts")
export class Account {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ length: 255 })
  name!: string;

  @Column({ length: 255 })
  surname!: string;

  @Column({ length: 255, unique: true })
  email!: string;

  @Column({ type: "varchar", length: 50, nullable: true })
  phone!: string | null;

  @Column({ name: "address_line1", length: 255 })
  addressLine1!: string;

  @Column({ name: "address_line2", type: "varchar", length: 255, nullable: true })
  addressLine2!: string | null;

  @Column({ length: 100 })
  city!: string;

  @Column({ length: 20 })
  postcode!: string;

  @Column({ length: 100 })
  country!: string;

  @Column({ type: "integer", default: 0 })
  balance!: number;

  @Column({ length: 3, default: "EUR" })
  currency!: string;

  @CreateDateColumn({ name: "created_at" })
  createdAt!: Date;

  @UpdateDateColumn({ name: "updated_at" })
  updatedAt!: Date;
}
